### ISSUE TITLE
fix(command/generate): directive selector should be camelcase

### ISSUE THREAD

**Issue by PascalPrecht** [angular/angular-cli#424]
Generating a directive using:

```
$ ng g directive my-directive
```

Generates a directive that looks like this:

```
@Directive({
  selector: 'my-directive',
  providers: [],
  host: {},

})
class MyDirective {}
```

There are a two things that need a fix IMO:

1. directive selectors should be camelcase, so instead of `my-directive` I should get `myDirective`
2. `providers` and `host` are not necessarily needed maybe we can simply remove those?
3. If we can't remove them, then the trailing comma and newline should be at least removed

**Comment by Brocco** [comment]
This will be included with the style-guide alignment PR I'm working on, so keep an eye out for that.

When running `ng g directive myDir` the selector should be `myDir`
However when running `ng g directive my-dir` should the selector be `my-dir` or `myDir`

//CC @IgorMinar @johnpapa

**Comment by johnpapa** [comment]
good question ... not sure. `mySillyDirectiveName` would be ideal .... but I think it is very reasonable to make it `mySiLyDiRECTiveName`

**Comment by johnpapa** [comment]
I agree with both of @PascalPrecht 's points too.

**Comment by Brocco** [comment]
For the initial I will implement it as what you provide as the argument will be the selector used in the directive this way the dev can explicitly decide on what the name will be.

**Comment by PascalPrecht** [comment]
@Brocco tbh I haven't thought of simply doing `ng g directive camelCase`. This would totally do it.

**Comment by Brocco** [comment]
@PascalPrecht no it won't, the current version will output the dasherized name see [here](https://github.com/angular/angular-cli/blob/4fa8de5867e3c3ade26ec0097853f581b3b9ddde/addon/ng2/blueprints/directive/files/__path__/__name__.ts#L5)

**Comment by PascalPrecht** [comment]
Oh then I misunderstood this. Thought this was already the case. Then I'm looking forward to this! :)

**Comment by Brocco** [comment]
If you want you can play with the branch I'm working on should have the full PR ready in the next few days

https://github.com/Brocco/angular-cli/tree/style-guide

**Comment by Brocco** [comment]
This issue has been addressed via PR #427 selectors for directives will not adjust the name entered via the CLI (i.e. `ng g directive FoObAr` will have a selector: `FoObAr`)

**Comment by JavascriptMick** [comment]
Angular style guide (https://angular.io/styleguide) still recommends lower camel case
Directive Selectors

Style 02-06

Do Use lower camel case for naming the selectors of our directives.
